internal/app/go-filecoin/node/message.go

```diff
@@ -8,6 +8,13 @@ import (
 )
 
 func (node *Node) processMessage(ctx context.Context, pubSubMsg pubsub.Message) (err error) {
+	sender := pubSubMsg.GetSender()
+
+	// ignore messages from self
+	if sender == node.Host().ID() {
+		return nil
+	}
```

Comment:
What necessitated this change?

Reply:
The lack of this check caused a very trick race in the mpool remove test.  This guard is what the messaging system appears to assume.  Otherwise we add to the mpool twice with difficult to capture timing semantics.